local_chainalysis.go

```diff
@@ -0,0 +1,78 @@
+package chainalysis
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/0xsequence/go-sequence/lib/prototyp"
+	"github.com/rs/zerolog/log"
+)
+
+const sanctionedAddressesSource = "https://raw.githubusercontent.com/0xsequence/chainalysis/initial-version/index/sanctioned_addresses.json"
+
+type sanctionedAddressEvent struct {
+	BlockNum  uint64          `json:"blockNum"`
+	BlockHash string          `json:"blockHash"`
+	Addrs     []prototyp.Hash `json:"addrs"`
+}
+
+type localChainalysis struct {
+	mu                  sync.RWMutex
+	SanctionedAddresses map[string]struct{}
+}
+
+func NewLocalChainalysis(ctx context.Context) (Chainalysis, error) {
+	sanctionedAddresses := map[string]struct{}{}
+
+	sanctionedAddressEvents, err := fetchSanctionedAddressEventsFromSource(sanctionedAddressesSource)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, event := range sanctionedAddressEvents {
+		for _, addr := range event.Addrs {
+			sanctionedAddresses[addr.String()] = struct{}{}
+		}
+	}
+
+	lc := &localChainalysis{
+		SanctionedAddresses: sanctionedAddresses,
+	}
+
+	go lc.fetcher(ctx)
```

Comment:
can you move this to a .Run() and .Stop() pattern like we do in other places..? see https://github.com/0xsequence/go-tokendirectory/blob/master/tokendirectory.go or more modern/recent databeat.. https://github.com/horizon-games/go-databeat/blob/master/databeat.go#L126